src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -289,9 +289,14 @@ public Task SendAsync(Message message)
 
         internal async Task<AmqpResponseMessage> ExecuteRequestResponseAsync(AmqpRequestMessage amqpRequestMessage)
         {
+            RequestResponseAmqpLink requestResponseAmqpLink = null;
             AmqpMessage amqpMessage = amqpRequestMessage.AmqpMessage;
             TimeoutHelper timeoutHelper = new TimeoutHelper(this.OperationTimeout, true);
-            RequestResponseAmqpLink requestResponseAmqpLink = await this.RequestResponseLinkManager.GetOrCreateAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
+
+            if (!this.RequestResponseLinkManager.TryGetOpenedObject(out requestResponseAmqpLink))
```

Comment:
if (!this.RequestResponseLinkManager.TryGetOpenedObject(out **var** requestResponseAmqpLink))
to remove unnecessary `RequestResponseAmqpLink requestResponseAmqpLink = null;`